src/main/java/com/github/manolo8/darkbot/extensions/plugins/PluginUpdater.java

```diff
@@ -0,0 +1,289 @@
+package com.github.manolo8.darkbot.extensions.plugins;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.gui.plugins.UpdateProgressBar;
+import com.github.manolo8.darkbot.gui.utils.Popups;
+import com.github.manolo8.darkbot.utils.Time;
+
+import javax.swing.*;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+//todoo i18n
+public class PluginUpdater {
+
+    private Date lastChecked = new Date();
+
+    public final List<Plugin> UPDATES = new ArrayList<>();
+
+    private final PluginHandler pluginHandler;
+
+    private UpdateProgressBar mainProgressBar;
+    private final List<PluginProgress> pluginProgresses = new ArrayList<>();
+
+    private static class PluginProgress {
+        private final Plugin plugin;
+        private final UpdateProgressBar progressBar;
+        private final JLabel progressLabel;
+
+        PluginProgress(Plugin plugin, UpdateProgressBar progressBar, JLabel progressLabel) {
+            this.plugin = plugin;
+            this.progressBar = progressBar;
+            this.progressLabel = progressLabel;
+        }
+
+        void setValue(int val) {
+            progressBar.setValue(val);
+        }
+
+        void setText(String text) {
+            progressLabel.setText(text);
+        }
+    }
+
+    public PluginUpdater(Main main) {
+        this.pluginHandler = main.pluginHandler;
+    }
+
+    public void scheduleUpdateChecker() {
+        Timer timer = new Timer();
+        timer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                checkUpdates();
+            }
+        }, 0, Time.DAY);
+    }
+
+    public void add(Plugin plugin, UpdateProgressBar progressBar, JLabel progressLabel) {
+        pluginProgresses.add(new PluginProgress(plugin, progressBar, progressLabel));
+    }
+
+    public void clear() {
+        pluginProgresses.clear();
+    }
+
+    public void setMainProgressBar(UpdateProgressBar progressBar) {
+        mainProgressBar = progressBar;
+    }
+
+    public boolean hasAvailableUpdates() {
+        return UPDATES.stream().anyMatch(pl -> pl.getUpdateStatus() == Plugin.UpdateStatus.AVAILABLE);
+    }
+
+    public boolean hasNoUpdates() {
+        return UPDATES.isEmpty();
+    }
+
+    public Date getLastChecked() {
+        return lastChecked;
+    }
+
+    public void checkUpdates() {
```

Comment:
This needs to be an async method, start a SwingWorker to search updates, and disable the button that allows searching while already running